SteamKit2/SteamKit2/Util/Win32Helpers.cs

```diff
@@ -1,258 +1,168 @@
-				public uint Size;
-			}
-
-			public struct STORAGE_DEVICE_DESCRIPTOR
-			{
-				public uint Version;
-				public uint Size;
-				public byte DeviceType;
-				public byte DeviceTypeModifier;
-				[MarshalAs(UnmanagedType.I1)]
-				public bool RemovableMedia;
-				[MarshalAs(UnmanagedType.I1)]
-				public bool CommandQueueing;
-				public uint VendorIdOffset;
-				public uint ProductIdOffset;
-				public uint ProductRevisionOffset;
-				public uint SerialNumberOffset;
-				public int StorageBusType;
-				public uint RawPropertiesLength;
-				[MarshalAs( UnmanagedType.ByValArray, SizeConst = 1 )]
-				public byte[] RawDeviceProperties;
-			}
-
-#pragma warning restore 0649
-
-			public static uint IOCTL_VOLUME_GET_VOLUME_DISK_EXTENTS = 0x00560000;
-			public static uint IOCTL_STORAGE_QUERY_PROPERTY = 0x002D1400;
-			public static int StorageDeviceProperty = 0;
-			public static int PropertyStandardQuery = 0;
-
-			[LibraryImport( "kernel32.dll", SetLastError = true )]
-			[return: MarshalAs( UnmanagedType.Bool )]
-			public static partial bool DeviceIoControl( SafeHandle hDevice, uint dwIoControlCode, IntPtr lpInBuffer, uint nInBufferSize, IntPtr lpOutBuffer, uint nOutBufferSize, out uint lpBytesReturned, IntPtr lpOverlapped );
-
-			#endregion
-		}
-	}
+    [SupportedOSPlatform( "windows5.1.2600" )]
+    static partial class Win32Helpers
+    {
+        public static string? GetBootDiskSerialNumber()
+        {
+            try
+            {
+                var bootDiskNumber = GetBootDiskNumber();
+                var serialNumber = GetPhysicalDriveSerialNumber( bootDiskNumber );
+                return serialNumber;
+            }
+            catch
+            {
+                return null;
+            }
+        }
+
+        static string GetBootDiskLogicalVolume()
+        {
+            var volume = Path.GetPathRoot( Environment.SystemDirectory.AsSpan() ).TrimEnd( Path.DirectorySeparatorChar );
+
+            if ( volume.Length == 0 )
+            {
+                throw new InvalidOperationException( "Could not determine system drive letter." );
+            }
+
+            return volume.ToString();
+        }
+
+        static unsafe uint GetBootDiskNumber()
+        {
+            using var handle = PInvoke.CreateFile(
+                $@"\\.\{GetBootDiskLogicalVolume()}",
+                0,
+                FILE_SHARE_MODE.FILE_SHARE_READ | FILE_SHARE_MODE.FILE_SHARE_WRITE,
+                null,
+                FILE_CREATION_DISPOSITION.OPEN_EXISTING,
+                FILE_FLAGS_AND_ATTRIBUTES.FILE_ATTRIBUTE_NORMAL,
+                null
+            );
+
+            if ( handle == null || handle.IsInvalid )
+            {
+                throw new FileNotFoundException( "Unable to open boot volume." );
+            }
+
+            var extents = new VOLUME_DISK_EXTENTS();
+
+            if ( !PInvoke.DeviceIoControl(
+                handle,
+                PInvoke.IOCTL_VOLUME_GET_VOLUME_DISK_EXTENTS,
```

Comment:
I would be tempted to `using static PInvoke;` and have this code look a little more like the C/C++ code it is emulating.